Register Page and Login Page "Hello World" options created.

eneadodi opened the pull request:
Totally forgot the naming conventions for branches too. Was afraid of deleting branches because I didn't know if it was remove my progress (github still a bit of a black box)
Anyways, The app redirects to register. Once register click button to get send to login. 
None of the information in the textboxes though ever actually gets saved...

**In PR:**
- Signup page
- Link on login page redirects to /signup
- Fixed node-fetch vulnerability

![LoginRegisterUIFinal](https://user-images.githubusercontent.com/51762139/94063768-9857fd80-fdb6-11ea-8542-9866bbede97f.gif)

Commit:
Trying to link register to login

Commit:
Register and Login Page set up. Link from Register to Login setup

Enoch-Hsiao commented on client_folder_react_app/ethisim/src/App.js:
```diff
@@ -6,22 +6,22 @@ import {
     <Router>
         <Route exact path="/">
-          <Redirect to="/login" />
+          <Redirect to="/register_account" />
```
I would rename register_account to /signup or /registration. Probably want to first redirect to login page first (rather than registration). And add a hyper link (below password box in /login, "Need to register an account?" that redirects to /signup); this can be done in another pr.

Enoch-Hsiao commented on client_folder_react_app/ethisim/src/pages/register_account.js:
```diff
@@ -0,0 +1,91 @@
+  Button,
+  TextField,
+  FormControlLabel,
+  Checkbox,
```
Remove FormControlLabel and CheckBox, unused imports.

Enoch-Hsiao commented on client_folder_react_app/ethisim/src/pages/register_account.js:
```diff
@@ -0,0 +1,91 @@
+            fullWidth
+            id="first_name"
+            label="first_name"
+            name="first name"
```
Label as "First Name" rather than "first_name". Same for last name.

Enoch-Hsiao commented on client_folder_react_app/ethisim/src/pages/register_account.js:
```diff
@@ -0,0 +1,91 @@
+            autoFocus
+          />
+          <Button
+          component={Link} to={"/login"}
```
Nice! We can implement a hyperlink to link to /register in the /login page using component={Link} to={"/login"} . But maybe add a comment somewhere that functionality for "Register Account!" is a "TODO" to be implemented. We don't want the "Register Account!" button to redirect to /login.

Enoch-Hsiao commented on client_folder_react_app/ethisim/src/pages/register_account.js:
```diff
@@ -0,0 +1,91 @@
+            label="last_name"
+            name="last name"
+            autoComplete="last name"
+            autoFocus
```
I would add text fields for username, password at the very least.

Enoch-Hsiao commented:
Reading data values from text-field box:
https://stackoverflow.com/questions/29791721/how-get-data-from-material-ui-textfield-dropdownmenu-components
https://material-ui.com/api/text-field/

Enoch-Hsiao committed:
Change 'Register Account' to 'Signup', added username and password boxes to signup page, removed unused imports, added hyperlink in login page 'Need to register an account?' to link to /signup

Enoch-Hsiao committed:
Change naming of 'Register' to 'Signup' and 'Create' for consistency

Enoch-Hsiao committed:
Code style, updated dependencies, fixed node-fetch vulnerability

TyGriffin115 committed:
20-Make email textbox for registration

Not sure if the ordering of

first name
last name
email address
username
password
//confirm password

is going to be our final order of the registration page.
TextBox for email address has been added.

TyGriffin115 committed:
21- Made Confirm Password Textbox

Added the Confirm Textbook to signup page

HerlinRijo reviewed (dismissed):
Looks good!

powellsl99 commented on client_folder_react_app/ethisim/src/pages/signup.js:
```diff
@@ -0,0 +1,134 @@
+import Copyright from '../components/copyright';
+
+export default function Signup() {
+  const useStyles = makeStyles((theme) => ({
```
*Nitpick* Generally useStyles goes above / outside of the functional component. Another way some people prefer is to make a directory for each component, which includes the component file, the makeStyles file, and the test file

powellsl99 commented on client_folder_react_app/ethisim/src/pages/signup.js:
```diff
@@ -0,0 +1,134 @@
+            label="Last Name"
+            name="last_name"
+            autoComplete="last_name"
+            autoFocus
```
autofocus should only be in one component (the first textfield)

powellsl99 commented on client_folder_react_app/ethisim/src/pages/signup.js:
```diff
@@ -0,0 +1,134 @@
+            label="Email Address"
+            name="email_address"
+            autoComplete="email_address"
+            autoFocus
```
remove autofocus here

powellsl99 commented on client_folder_react_app/ethisim/src/pages/signup.js:
```diff
@@ -0,0 +1,134 @@
+            label="Username"
+            name="username"
+            autoComplete="username"
+            autoFocus
```
remove autofocus here

powellsl99 commented on client_folder_react_app/ethisim/src/pages/signup.js:
```diff
@@ -0,0 +1,134 @@
+            label="Password"
+            name="password"
+            autoComplete="password"
+            autoFocus
```
remove autofocus here

powellsl99 commented on client_folder_react_app/ethisim/src/pages/signup.js:
```diff
@@ -0,0 +1,134 @@
+            label="Confirm Password"
+            name="confirm_password"
+            autoComplete="confirm_password"
+            autoFocus
```
remove autofocus here

powellsl99 commented on client_folder_react_app/ethisim/src/App.js:
```diff
@@ -2,26 +2,20 @@ import React from 'react';
 import './App.css';
```
I don't think this is being used anymore

powellsl99 commented on client_folder_react_app/ethisim/src/App.js:
```diff
@@ -2,26 +2,20 @@ import React from 'react';
-          </Route>
-        </Switch>
+        <Route path = "/login" component={Login}/>
+        <Route path="/signup" component={Signup}></Route>
```
Should keep style consistency, and either explicitly include the `</Route>` on both, or neither

powellsl99 commented on client_folder_react_app/ethisim/src/pages/signup.js:
```diff
@@ -0,0 +1,134 @@
+            autoComplete="password"
+            autoFocus
+          />
+           <TextField
```
*Doesn't have to be for this PR* : once you link the inputs to state, you should do some sort of validation to make sure the user knows if the passwords aren't the same

Enoch-Hsiao replied:
I think we will have another pr where we add store values from the input boxes and check for passwords.

powellsl99 reviewed (changes requested):
This is a great PR! A few small changes and it should be good to go! You may also want to look into Prettier+Husky at some point soon. It is easy to set up and will make sure you are all using the same coding conventions

Enoch-Hsiao committed:
Addressed PR comments

Enoch-Hsiao committed:
put 'Need to create an account?' and checkbox for remember me in a grid

Enoch-Hsiao committed:
remove package.lock.json file outside of react app repository

powellsl99 reviewed (approved):
Looks good on my end. Great job all!!
